### PR TITLE
Plugin extensions: Add e2e tests

### LINKED ISSUE #89024: Plugin extensions: Add e2e tests
**Why is this needed**:

[Add a description of the problem the feature is trying to solve.]

**What would you like to be added**:

[Add a brief description of what the feature or enhancement should do.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

### PR DESCRIPTION
**Pull request by sunker**
**What is this feature?**

This PR adds Playwright e2e tests that verifies that app plugins can define extension points and configure link extensions (path and onclick) using the latest Grafana runtime. 

To be able to test this, there needs to be apps with extensions (and extension points) installed on the instance that is being targeted in the e2e tests. For now, I've just copied the bundles from [app-with-extension-point](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-extension-point) and [app-with-extensions ](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-extensions) to the plugins directory. This is not ideal as it would require us to manually update the bundles whenever the examples are updated. However, I think it will still be useful as these example plugins use the now deprecated extensions APIs (configure, configureExtensionLink, getPluginExtensions etc), and we need to support them for roughly one more year.  

The e2e tests and the test dashboards are copied from the examples repo. 

**Why do we need this feature?**

To verify that legacy plugin extensions APIs are still working as expected. 

**Who is this feature for?**

Plugin authors

**Which issue(s) does this PR fix?**:

Fixes #89024

**Special notes for your reviewer:**

This change makes running e2e tests locally slightly more complicated, as it requires you to have these app plugins installed and provisioned on your local instance. As the only purpose of these apps is e2e testing and not development, I don't want to create them with the devenv scripts. So to be able to get these tests to work locally, one need to start the grafana server the [same way](https://github.com/grafana/grafana/blob/extensions/add-e2e-tests/scripts/grafana-server/start-server) as in Drone. Otherwise, the apps won't be installed. I've added a new script, `e2e:playwright:server`, that starts up the server and runs the tests.

### CODE REVIEW

**Review by mckn — APPROVED** [review]
Great work on this!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -223,6 +223,7 @@
 /devenv/local-npm/ @grafana/frontend-ops
 /devenv/vscode/ @grafana/frontend-ops
 /devenv/setup.sh @grafana/grafana-backend-services-squad
+devenv/plugins.yaml @grafana/plugins-platform-frontend
```
Comment (leventebalogh) [inline .github/CODEOWNERS:226]: Just double-checking here - I guess we don't need the leading slash in front of `devenv/plugins.yaml` right?

File: package.json [hunk 2]
```diff
@@ -18,6 +18,7 @@
     "e2e:enterprise:dev": "./e2e/start-and-run-suite enterprise dev",
     "e2e:enterprise:debug": "./e2e/start-and-run-suite enterprise debug",
     "e2e:playwright": "yarn playwright test",
+    "e2e:playwright:server": "./e2e/plugin-e2e/start-and-run-suite",
```
Comment (leventebalogh) [inline package.json:21]: Should we add a note about this command to the [docs](https://github.com/grafana/grafana/blob/main/contribute/style-guides/e2e-plugins.md#commands) as well?

**Review by leventebalogh — APPROVED** [review]
Nice one 👏

**Comment by leventebalogh** [comment]
Just thinking out loud: maybe it would make sense to add an extra README next to the bundled plugins, that gives a short documentation on how these should be updated (in case someone wants to add a new extension / extension point).

File: .github/CODEOWNERS (hunk 1 above)
Comment (sunker) [reply .github/CODEOWNERS]: oh. looks like the build passed, so the file was picked up. but I'm changing this. thanks for finding this!

File: package.json (hunk 2 above)
Comment (sunker) [reply package.json:21]: Good point! Added to the e2e-plugins.md docs and to the developer-guide.

**Comment by sunker** [comment]
> Just thinking out loud: maybe it would make sense to add an extra README next to the bundled plugins, that gives a short documentation on how these should be updated (in case someone wants to add a new extension / extension point).

Yes this is a good point! Added a readme (see [this](https://github.com/grafana/grafana/pull/89048/commits/bb894d0ae6b4a8bf704d437018ea9fac33bd5518) commit). @academo you may want to have a peek at the readme since you introduced this folder a while back).

**Pull request merged**